lib/chef/resource.rb

```diff
@@ -599,10 +599,18 @@ def run_action(action, notification_type = nil, notifying_resource = nil)
         with_umask do
           provider_for_action(action).run_action
         end
+
       rescue StandardError => e
+        error = e.message.sub /\s*\(.+\)$/, ""
         if ignore_failure
           logger.error("#{custom_exception_message(e)}; ignore_failure is set, continuing")
           events.resource_failed(self, action, e)
+        # Added condition for http request
+        # remote_file resource is actually calling a HTTP resource
+        # HTTP resource already has retry_delay/retries
+        elsif e.class == SocketError && http_request_errors.include?(error)
+          events.resource_failed(self, action, e)
+          raise e.class, e.message
```

Comment:
i think all this code can now get removed, the default should be to not retry here, but to manage HTTP retries in the Chef::HTTP class itself, any errors that escape that class should get exposed directly here.

Reply:
Updated this, please have a look.